apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -52,22 +100,84 @@ router.post(
     if (!config.openAiApiKey || !config.openAiOrganization) {
       throw new error.HttpStatusError(403, 'Not implemented (feature not available)');
     }
+
     const client = new OpenAI({
       apiKey: config.openAiApiKey,
       organization: config.openAiOrganization,
     });
 
-    if (req.body.__action === 'generate_question') {
-      const jobSequenceId = await generateQuestion(
+    if (req.body.__action === 'ai_generate') {
+      const jobSeq = await generateQuestion(
         client,
         res.locals.course ? res.locals.course.id : undefined,
         res.locals.authn_user.user_id,
         req.body.prompt,
       );
-      res.redirect(res.locals.urlPrefix + '/jobSequence/' + jobSequenceId);
+
+      const data = jobSeq.jobData;
+      if (data) {
+        res.send(
+          GenerationResults(data.data.html, data.data.python, jobSeq.jobSequenceId, res.locals),
+        );
+      } else {
+        throw new error.HttpStatusError(500, `Job Sequence ${jobSeq.jobSequenceId} failed.`);
+      }
     } else if (req.body.__action === 'sync_context_documents') {
       const jobSequenceId = await syncContextDocuments(client, res.locals.authn_user.user_id);
       res.redirect('/pl/jobSequence/' + jobSequenceId);
+    } else if (req.body.__action === 'ai_regen') {
+      const genJobs = await selectJobsByJobSequenceId(req.body.unsafe_sequence_job_id);
+      if (
+        genJobs.length !== 1 ||
+        !genJobs[0]?.course_id ||
+        !idsEqual(genJobs[0]?.course_id, res.locals.course.id)
+      ) {
+        throw new error.HttpStatusError(
+          403,
+          `Course for job sequence ${req.body.unsafe_sequence_job_id} does not match the current course.`,
```

Comment:
We can just say something generic like "Job sequence #### not found".

Reply:
changed in e0331beaf9d587990dcda087d800feedc786fc87.